AdvLoggerPkg/Library/AdvancedLoggerLib/MmCoreArm/AdvancedLoggerLib.c

```diff
@@ -114,12 +114,20 @@ AdvancedLoggerGetLoggerInfo (
     // If this is the first time for MM core to get here, the memory attributes of this module
     // may not be fully set yet. Thus set the memory for global variables attributes to RW first.
     Address = ALIGN_VALUE ((EFI_PHYSICAL_ADDRESS)(UINTN)&mInitialized - EFI_PAGE_SIZE + 1, EFI_PAGE_SIZE);
-    Status  = ArmSetMemoryRegionNoExec (Address, EFI_PAGE_SIZE);
-    Status  = ArmClearMemoryRegionReadOnly (Address, EFI_PAGE_SIZE);
+    Status  = ArmSetMemoryAttributes (

```

Comment:
I think we should better use `MmuSetAttributes (Address, EFI_PAGE_SIZE, EFI_MEMORY_XP)` here. And include the "<Library/MmuLib.h>" instead.

Also, we also need to clear the read only flag, right?

Reply:
I agree on using MmuLib, to get off ARM specific libs, Will make that change.

Side rant, I'm not a fan of this whole "clear/set" semantic when it comes to attributes. A caller will always want/need a specific set of attributes for the use case of the buffer, so it sees really odd to me to allow a caller to only ask about specific attributes. I believe they should set to an explicit combination each time. _steps off soapbox_

Reply:
I think this is to prevent the caller always need to retrieve the current attribute? But I agree that this pattern is confusing when we want to replace the current page attributes with ours.

Reply:
My point is, why do we care about the current attributes? If we have a buffer for a purpose, we know what all 3 attributes should be, so we should set them so what we want despite what they currently are. but yeah, not relevant here.